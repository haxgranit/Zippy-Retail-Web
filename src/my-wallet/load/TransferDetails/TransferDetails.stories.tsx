import { ComponentStory, ComponentMeta } from '@storybook/react';
import { BrowserRouter } from 'react-router-dom';
import React from 'react';
import TransferDetails from './TransferDetails';

export default {
  title: 'my-wallet/load/TransferDetails',
  component: TransferDetails,
} as ComponentMeta<typeof TransferDetails>;

const Template: ComponentStory<typeof TransferDetails> = () => (
  <BrowserRouter>
    <TransferDetails />
  </BrowserRouter>
);

export const Default = Template.bind({});
Default.args = {};
Default.parameters = {};
