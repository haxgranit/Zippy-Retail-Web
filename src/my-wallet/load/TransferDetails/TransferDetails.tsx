import { useMsal } from '@azure/msal-react';
import { useEffect, useState } from 'react';
import { Button } from 'react-bootstrap';
import { useLocation, useNavigate } from 'react-router-dom';
import Api from '../../../api';
import PageContainer from '../../../common/PageContainer';
import { FundingSourceTransaction } from '../../../constants/type/FundingSourceTransaction';

type LocationState = {
  id: number,
  sourceId: number,
  isCredit: boolean
};

export default function TransferDetails() {
  const { instance, accounts } = useMsal();
  const [transferDetails, settransferDetails] = useState<FundingSourceTransaction | null>(null);
  const state = useLocation().state as LocationState;
  const [dateTime, setDateTime] = useState<{ date: string, time:string }>({ date: '', time: '' });
  const navigate = useNavigate();
  useEffect(() => {
    new Api(instance, accounts[0])
      .getFundingSourceTransaction(state.id, state.sourceId)
      .then((data: FundingSourceTransaction) => {
        settransferDetails(data);
        const today = new Date(data?.createdDate);
        const month = today.toLocaleString('default', { month: 'short' });
        setDateTime({
          date: `${month} ${today.getDate()}, ${today.getFullYear()}`,
          time: today.toLocaleString('en-US', { hour: 'numeric', minute: 'numeric', hour12: true }),
        });
      })
      .catch((error) => {
        // eslint-disable-next-line no-console
        console.log('error', error);
      });
  }, []);

  return (
    <PageContainer title="Your Wallet" subTitle="Made Fun With Zippy!">
      <div className="body">
        {transferDetails?.status === 'pending' ? (
          <div className="title pending">
            Transfer Pending
          </div>
        ) : (
          <div className="title complete">
            Transfer Completed
          </div>
        )}

        <div className="transaction-detail">
          <div className="detail-wrap">
            <span>Transfer Type</span>

            <p>Zippy Wallet Load</p>
          </div>
          <div className="detail-wrap">
            <span>Transfer Date</span>

            <span>{dateTime.date}</span>
          </div>

          <div className="detail-wrap">
            <span>Transfer Time (UTC)</span>

            <span>{dateTime.time}</span>
          </div>

          <div className="detail-wrap">
            <span>Transfer Amount</span>

            <p style={{ color: '#0D6EFD' }}>{ `$${transferDetails?.amount ?? ''}`}</p>
          </div>

          <div className="detail-wrap">
            <span>Reference Number (Keep For Your Records)</span>

            <p>{transferDetails?.id}</p>
          </div>
        </div>
      </div>
      <div className="action">
        <Button
          className="zippy-btn"
          onClick={() => navigate(`/my-wallet/${state?.isCredit ? 'get' : 'load'}`)}
        >
          {state?.isCredit ? 'GET' : 'LOAD'}
        </Button>
      </div>
    </PageContainer>
  );
}
