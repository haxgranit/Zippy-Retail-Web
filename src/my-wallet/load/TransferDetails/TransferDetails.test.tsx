import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import TransferDetails from './TransferDetails';

const ReactTestRenderer = require('react-test-renderer');

describe('TransactionComplete Component', () => {
  it('matches the snapshot', () => {
    const tree = ReactTestRenderer.create(
      <BrowserRouter>
        <TransferDetails />
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
