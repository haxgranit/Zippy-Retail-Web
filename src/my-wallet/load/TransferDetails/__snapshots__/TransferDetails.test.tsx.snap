// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionComplete Component matches the snapshot 1`] = `
<div
  className=""
>
  <div
    className="zippy-cash-container-wrap"
  >
    <div
      className="backdrop "
    />
    <div
      className="zippy-cash-container"
    >
      <div
        className="zippy-cash-container-header"
      >
        <h1
          className="title"
        >
          Your Wallet
        </h1>
        <h2
          className="catch"
        >
          Made Fun With Zippy!
        </h2>
      </div>
      <div
        className="zippy-cash-container-body"
      >
        <div
          className="body"
        >
          <div
            className="title complete"
          >
            Transfer Completed
          </div>
          <div
            className="transaction-detail"
          >
            <div
              className="detail-wrap"
            >
              <span>
                Transfer Type
              </span>
              <p>
                Zippy Wallet Load
              </p>
            </div>
            <div
              className="detail-wrap"
            >
              <span>
                Transfer Date
              </span>
              <span>
                
              </span>
            </div>
            <div
              className="detail-wrap"
            >
              <span>
                Transfer Time (UTC)
              </span>
              <span>
                
              </span>
            </div>
            <div
              className="detail-wrap"
            >
              <span>
                Transfer Amount
              </span>
              <p
                style={
                  Object {
                    "color": "#0D6EFD",
                  }
                }
              >
                $
              </p>
            </div>
            <div
              className="detail-wrap"
            >
              <span>
                Reference Number (Keep For Your Records)
              </span>
              <p />
            </div>
          </div>
        </div>
        <div
          className="action"
        >
          <button
            className="zippy-btn btn btn-primary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            LOAD
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
